kubeflow/trainer/backends/kubernetes/utils.py

```diff
@@ -262,10 +259,43 @@ def get_script_for_python_packages(
         """
     )
 
+    # Decide whether to append --user at container runtime:
+    # - Always for MPI
+    # - Or when running as non-root and not inside a virtual environment
+    runtime_logic = textwrap.dedent(
+        f"""\
+MPI_FLAG={1 if is_mpi else 0}
```

Comment:
I am curious wether it would be easier to just run `pip install --user` if that fails, run `pip install` without `--user`?
Something like:
```
pip install --user .... || pip install ....
```